<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>SuSiE vs. FINEMAP in an example where the causal SNPs have relatively large effects</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">finemap</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Overview</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/stephenslab/finemap">source</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SuSiE vs. FINEMAP in an example where the causal SNPs have relatively large effects</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>finemap/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed1code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(1)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed1code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(1)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfinemaptree38a8ed50c67dd1e656bb95bfa089816d14e46d21targetblank38a8ed5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/stephenslab/finemap/tree/38a8ed50c67dd1e656bb95bfa089816d14e46d21" target="_blank">38a8ed5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomstephenslabfinemaptree38a8ed50c67dd1e656bb95bfa089816d14e46d21targetblank38a8ed5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/stephenslab/finemap/tree/38a8ed50c67dd1e656bb95bfa089816d14e46d21" target="_blank">38a8ed5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    data/small_data_11.ld_refout_file.refout.ld
    Ignored:    data/small_data_11_sim_gaussian_pve_n_8_get_sumstats_n_1.ld_sample_n_file.in_n.ld
    Ignored:    output/sim1.config
    Ignored:    output/sim2.config

Unstaged changes:
    Modified:   output/sim1.cred1
    Modified:   output/sim1.cred2
    Modified:   output/sim1.log_sss
    Modified:   output/sim1.snp
    Modified:   output/sim2.bf5
    Modified:   output/sim2.cred5
    Modified:   output/sim2.log_sss
    Modified:   output/sim2.snp

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/large_effect.Rmd</code>) and HTML (<code>docs/large_effect.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/38a8ed50c67dd1e656bb95bfa089816d14e46d21/analysis/large_effect.Rmd" target="_blank">38a8ed5</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-10
</td>
<td>
workflowr::wflow_publish(&quot;large_effect.Rmd&quot;, view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/e597078613eccec56421d2eb742c8e6925472770/docs/large_effect.html" target="_blank">e597078</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-10
</td>
<td>
Added finemap results to large_effect example.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/c3f0aa36bdb09f42e9c4d4f00d2dc5b594d2faa0/analysis/large_effect.Rmd" target="_blank">c3f0aa3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-10
</td>
<td>
workflowr::wflow_publish(&quot;large_effect.Rmd&quot;, view = FALSE, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/6c863a1f9725be8b803c3d2ffc96e9d3d877e8a1/analysis/large_effect.Rmd" target="_blank">6c863a1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-10
</td>
<td>
Working on adding finemap analysis to large_effect example.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/f40a3371cf2fddff478035de02f3255f0c5c0650/docs/large_effect.html" target="_blank">f40a337</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-09
</td>
<td>
Adjusted plot dims slightly.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/9dc1f67b5012476fdf01e5d1346365806f5d557c/analysis/large_effect.Rmd" target="_blank">9dc1f67</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-09
</td>
<td>
workflowr::wflow_publish(&quot;large_effect.Rmd&quot;, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/37627a79ca9fa6005c154dd5592d5519e78a046b/docs/large_effect.html" target="_blank">37627a7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-09
</td>
<td>
Added susie-rss fit using out-of-sample LD.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/2c327c826cad7e1720f9354ae293d305d4cc319a/analysis/large_effect.Rmd" target="_blank">2c327c8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-09
</td>
<td>
workflowr::wflow_publish(&quot;large_effect.Rmd&quot;, verbose = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/7af3892991629521646cf3e08822e119fe2691d5/analysis/large_effect.Rmd" target="_blank">7af3892</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-08
</td>
<td>
Working on large_effect demo.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/cd367c3af16f9e8ad1516c57126683dad0110859/docs/large_effect.html" target="_blank">cd367c3</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-08
</td>
<td>
Added step to run susie with in-sample ld matrix.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/1f607ceb1aba2fa0e2e975a8b815f4042cd5ac88/analysis/large_effect.Rmd" target="_blank">1f607ce</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-08
</td>
<td>
workflowr::wflow_publish(&quot;large_effect.Rmd&quot;)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/f3b3f264be4280b9ab9a00a173bd42efe02e3c7d/analysis/large_effect.Rmd" target="_blank">f3b3f26</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-08
</td>
<td>
More improvements to large_effect demo.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/ba1823bf4db72c49cabda54f710e41768b00c9e0/analysis/large_effect.Rmd" target="_blank">ba1823b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-08
</td>
<td>
Working on large_effect demo.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/ba1823bf4db72c49cabda54f710e41768b00c9e0/docs/large_effect.html" target="_blank">ba1823b</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-08
</td>
<td>
Working on large_effect demo.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/e68ebf0916f6707ebb06f78ac384726205ff8e36/docs/large_effect.html" target="_blank">e68ebf0</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/5ba80d93ad231724f9721479fa516e16d752cf8c/analysis/large_effect.Rmd" target="_blank">5ba80d9</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-07
</td>
<td>
Small edit.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/stephenslab/finemap/5737016b5a2039bd931e88d3163091fa4325014d/docs/large_effect.html" target="_blank">5737016</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-07
</td>
<td>
First build of large_effect analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/4b1ca7f9ad44b7694751997f21c7b572e21c8dcf/analysis/large_effect.Rmd" target="_blank">4b1ca7f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-07
</td>
<td>
workflowr::wflow_publish(&quot;large_effect.Rmd&quot;)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/stephenslab/finemap/blob/eae747f621b791bdfd5a94a2fdd28065bb28a7f8/analysis/large_effect.Rmd" target="_blank">eae747f</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-07
</td>
<td>
Revised workflowr files.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In this small example drawn from our <a href="https://github.com/stephenslab/dsc_susierss">simulations</a>, we show that that <a href="http://www.christianbenner.com">FINEMAP</a> works well with an &quot;in-sample LD&quot; matrix—that is, a correlation matrix that was estimated using the same sample that was used to compute the single-SNP association statistics—but, can perform surprisingly poorly with an &quot;out-of-sample&quot; LD matrix. We have observed that this degradation in performance is unusual—specifically, when the effects of the causal SNPs are very large. Nonetheless, this example may be instructive. In contrast to FINEMAP, <a href="https://github.com/stephenslab/susieR">SuSiE</a> performs similarly well in this example with either the in-sample and out-of-sample LD matrix.</p>
<p>First, load some packages, and set the seed for reproducibility.</p>
<pre class="r"><code>library(data.table)
library(susieR)
set.seed(1)</code></pre>
<p>Load the summary data: the least-squares effect estimates <span class="math">\(\hat{\beta}_i\)</span> and their standard errors <span class="math">\(\hat{s}_i\)</span> for each SNP <span class="math">\(i\)</span>. Here we also compute the <em>z</em>-scores since SuSiE accepts <em>z</em>-scores as input.</p>
<pre class="r"><code>dat1 &lt;- readRDS(&quot;../data/small_data_11.rds&quot;)
dat3 &lt;- readRDS(&quot;../data/small_data_11_sim_gaussian_pve_n_8_get_sumstats_n_1.rds&quot;)
maf  &lt;- dat1$maf$in_sample
bhat &lt;- dat3$sumstats$bhat
shat &lt;- dat3$sumstats$shat
z    &lt;- bhat/shat</code></pre>
<p>In this simulation, 2 out of 1,001 SNPs affect the phenotype:</p>
<pre class="r"><code>dat2 &lt;- readRDS(&quot;../data/small_data_11_sim_gaussian_pve_n_8.rds&quot;)
b &lt;- drop(dat2$meta$true_coef)
vars &lt;- which(b != 0)
vars
# [1] 305 740</code></pre>
<div id="in-sample-ld" class="section level2">
<h2>In-sample LD</h2>
<p>Run SuSiE with the &quot;in-sample&quot; LD estimate:</p>
<pre class="r"><code>ldinfile &lt;- &quot;../data/small_data_11_sim_gaussian_pve_n_8_get_sumstats_n_1.ld_sample_n_file.in_n.ld&quot;
Rin &lt;- as.matrix(fread(ldinfile))
fit1 &lt;- susie_rss(z,Rin,n = 800,min_abs_corr = 0.1,refine = FALSE,
                 verbose = TRUE)
# HINT: If the in-sample LD matrix is available, we recommend calling susie_rss with the in-sample LD matrix, and setting estimate_residual_variance = TRUE
# HINT: For large R or large XtX, consider installing theRfast package for better performance.
# WARNING: XtX is not symmetric; forcing XtX to be symmetric by replacing XtX with (XtX + t(XtX))/2
# [1] &quot;objective:-1022.11016014142&quot;
# [1] &quot;objective:-1022.09485818746&quot;
# [1] &quot;objective:-1022.08938881341&quot;
# [1] &quot;objective:-1022.08631623581&quot;
# [1] &quot;objective:-1022.08453069756&quot;
# [1] &quot;objective:-1022.08341717686&quot;
# [1] &quot;objective:-1022.08267674839&quot;</code></pre>
<p>(Although the recommendation is to estimate the residual variance, to maintain consistency with the analysis using out-of-sample LD, we ignore this advice.)</p>
<p>SuSiE returns a single credible set (CS) containing a large number of strongly correlated SNPs. One of the SNPs in this CS is a (true) causal SNP.</p>
<pre class="r"><code>print(fit1$sets[c(&quot;cs&quot;,&quot;purity&quot;)])
# $cs
# $cs$L1
#  [1] 195 197 203 213 226 237 238 243 247 248 249 254 255 278 294 296 301 305 319
# [20] 325 351 371 380 381 389 390 393 405 420 421 422 424 427 434 435 436 437 438
# [39] 441 442 443 445 448 450 452 454 456 459 462 464 466 467 468 473 477 478 479
# [58] 483 484 485 486 487 488 489 490 492 493 497 503 504 512 520 535 552 554 555
# [77] 558 571
# 
# 
# $purity
#    min.abs.corr mean.abs.corr median.abs.corr
# L1    0.9827454     0.9993761       0.9999995</code></pre>
<p>Here's a visualization of this result. (In this plot, the CS is depicted by the light blue circles, and the two causal SNPs are drawn as red triangles.)</p>
<pre class="r"><code>par(mar = c(4,4,1,1))
cs1 &lt;- fit1$sets$cs$L1
plot(1:1001,fit1$pip,pch = 20,cex = 0.8,ylim = c(0,0.1),
     xlab = &quot;SNP&quot;,ylab = &quot;susie PIP&quot;)
points(cs1,fit1$pip[cs1],pch = 1,cex = 1,col = &quot;cyan&quot;)
points(vars,fit1$pip[vars],pch = 2,cex = 0.8,col = &quot;tomato&quot;)</code></pre>
<p><img src="figure/large_effect.Rmd/susie-in-3-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-susie-in-3-1">
Past versions of susie-in-3-1.png
</button>
</p>
<div id="fig-susie-in-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap/blob/e597078613eccec56421d2eb742c8e6925472770/docs/figure/large_effect.Rmd/susie-in-3-1.png" target="_blank">e597078</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap/blob/f40a3371cf2fddff478035de02f3255f0c5c0650/docs/figure/large_effect.Rmd/susie-in-3-1.png" target="_blank">f40a337</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap/blob/37627a79ca9fa6005c154dd5592d5519e78a046b/docs/figure/large_effect.Rmd/susie-in-3-1.png" target="_blank">37627a7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now let's try running FINEMAP on these same data:</p>
<pre class="r"><code>run_finemap &lt;- function (bhat, shat, maf, prefix) {
  p &lt;- length(b)
  dat &lt;- data.frame(rsid       = 1:p,
                    chromosome = rep(1,p),
                    position   = rep(1,p),
                    allele1    = rep(&quot;A&quot;,p),
                    allele2    = rep(&quot;C&quot;,p),
                    maf        = round(maf,digits = 6),
                    beta       = round(bhat,digits = 6),
                    se         = round(shat,digits = 6))
  outfile &lt;- paste(prefix,&quot;z&quot;,sep = &quot;.&quot;)
  masterfile &lt;- paste(prefix,&quot;master&quot;,sep = &quot;.&quot;)
  write.table(dat,outfile,quote = FALSE,col.names = TRUE,row.names = FALSE)
  out &lt;- system(paste(&quot;./finemap_v1.4.1_x86_64 --sss --log --n-causal-snps 5&quot;,
                      &quot;--in-files&quot;,masterfile),intern = TRUE)
  out &lt;- out[which(!grepl(&quot;Reading&quot;,out,fixed = TRUE))]
  out &lt;- out[which(!grepl(&quot;Computing&quot;,out,fixed = TRUE))]
  out &lt;- out[which(!grepl(&quot;evaluated&quot;,out,fixed = TRUE))]
  cat(out,sep = &quot;\n&quot;)
  return(invisible(out))
}
setwd(&quot;../output&quot;)
run_finemap(bhat,shat,maf,prefix = &quot;sim1&quot;)
setwd(&quot;../analysis&quot;)
# 
# |--------------------------------------|
# | Welcome to FINEMAP v1.4.1            |
# |                                      |
# | (c) 2015-2022 University of Helsinki |
# |                                      |
# | Help :                               |
# | - ./finemap --help                   |
# | - www.finemap.me                     |
# | - www.christianbenner.com            |
# |                                      |
# | Contact :                            |
# | - finemap@christianbenner.com        |
# | - matti.pirinen@helsinki.fi          |
# |--------------------------------------|
# 
# --------
# SETTINGS
# --------
# - dataset            : all
# - corr-config        : 0.95
# - n-causal-snps      : 5
# - n-configs-top      : 50000
# - n-conv-sss         : 100
# - n-iter             : 100000
# - n-threads          : 1
# - prior-k0           : 0
# - prior-std          : 0.05 
# - prob-conv-sss-tol  : 0.001
# - prob-cred-set      : 0.95
# 
# ------------
# FINE-MAPPING (1/1)
# ------------
# - GWAS summary stats               : sim1.z
# - SNP correlations                 : ../data/small_data_11_sim_gaussian_pve_n_8_get_sumstats_n_1.ld_sample_n_file.in_n.ld
# - Causal SNP stats                 : sim1.snp
# - Causal configurations            : sim1.config
# - Credible sets                    : sim1.cred
# - Log file                         : sim1.log_sss
# 
# 
- Updated prior SD of effect sizes : 0.05 0.0902 0.163 0.293 
# 
# - Number of GWAS samples           : 800
# - Number of SNPs                   : 1001
# - Prior-Pr(# of causal SNPs is k)  : 
#   (0 -&gt; 0)
#    1 -&gt; 0.583
#    2 -&gt; 0.291
#    3 -&gt; 0.097
#    4 -&gt; 0.0242
#    5 -&gt; 0.00483
# - Regional SNP heritability        : 0.246 (SD: 0.0278 ; 95% CI: [0.194,0.303])
# - Log10-BF of &gt;= one causal SNP    : 57.1
# - Post-expected # of causal SNPs   : 1.74
# - Post-Pr(# of causal SNPs is k)   : 
#   (0 -&gt; 0)
#    1 -&gt; 0.262
#    2 -&gt; 0.738
#    3 -&gt; 0
#    4 -&gt; 0
#    5 -&gt; 0
# - Run time                         : 0 hours, 0 minutes, 10 seconds</code></pre>
<p>FINEMAP predicts 2 causal SNPs (with 74% probability). The second FINEMAP credible set is a diffuse set with lots of uncorrelated SNPs, so we visualize the first CS only:</p>
<pre class="r"><code>par(mar = c(4,4,1,1))
finemap &lt;- read.table(&quot;../output/sim1.cred2&quot;,header = TRUE)
pip   &lt;- rep(0,1001)
cs1   &lt;- finemap$cred1
rows1 &lt;- which(!is.na(cs1))
cs1   &lt;- cs1[rows1]
pip[cs1] &lt;- pip[cs1] + finemap$prob1[rows1]
plot(1:1001,pip,pch = 20,cex = 0.8,ylim = c(0,0.1),
     xlab = &quot;SNP&quot;,ylab = &quot;finemap PIP&quot;)
points(cs1,pip[cs1],pch = 1,cex = 1,col = &quot;cyan&quot;)
points(vars,pip[vars],pch = 2,cex = 0.8,col = &quot;tomato&quot;)</code></pre>
<p><img src="figure/large_effect.Rmd/finemap-in-2-1.png" width="540" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-finemap-in-2-1">
Past versions of finemap-in-2-1.png
</button>
</p>
<div id="fig-finemap-in-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/stephenslab/finemap/blob/e597078613eccec56421d2eb742c8e6925472770/docs/figure/large_effect.Rmd/finemap-in-2-1.png" target="_blank">e597078</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2022-06-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This looks very similar to the SuSiE credible set, which is reassuring.</p>
</div>
<div id="out-of-sample-ld" class="section level2">
<h2>Out-of-sample LD</h2>
<p>Let's try running SuSiE and FINEMAP as before, except using an out-of-sample LD estimate.</p>
<pre class="r"><code>ldoutfile &lt;- &quot;../data/small_data_11.ld_refout_file.refout.ld&quot;
Rout &lt;- as.matrix(fread(ldoutfile))
fit2 &lt;- susie_rss(z,Rout,n = 800,min_abs_corr = 0.1,refine = FALSE,
                 verbose = TRUE)
# HINT: If the in-sample LD matrix is available, we recommend calling susie_rss with the in-sample LD matrix, and setting estimate_residual_variance = TRUE
# HINT: For large R or large XtX, consider installing theRfast package for better performance.
# WARNING: XtX is not symmetric; forcing XtX to be symmetric by replacing XtX with (XtX + t(XtX))/2
# [1] &quot;objective:-1022.15334027678&quot;
# [1] &quot;objective:-1022.1533319747&quot;</code></pre>
<p>As above, SuSiE returns a single CS containing one of the two causal SNPs.</p>
<pre class="r"><code>print(fit2$sets[c(&quot;cs&quot;,&quot;purity&quot;)])
# $cs
# $cs$L1
#  [1] 195 197 203 213 226 237 238 243 247 248 249 254 255 278 294 296 301 305 319
# [20] 325 351 371 380 381 389 390 393 405 420 421 422 424 427 434 435 436 437 438
# [39] 441 442 443 445 448 450 452 454 456 459 462 464 466 467 468 473 477 478 479
# [58] 483 484 485 486 487 488 489 490 492 493 497 503 504 512 520 535 552 554 555
# [77] 558 571
# 
# 
# $purity
#    min.abs.corr mean.abs.corr median.abs.corr
# L1    0.9759149     0.9986827        0.999996</code></pre>
<p>In fact, in this particular example the CS is exactly the same as the CS above obtained using the in-sample LD:</p>
<pre class="r"><code>all(fit1$sets$cs$L1 == fit2$sets$cs$L1)
# [1] TRUE</code></pre>
<p>Now let's run FINEMAP:</p>
<pre class="r"><code>setwd(&quot;../output&quot;)
run_finemap(bhat,shat,maf,prefix = &quot;sim2&quot;)
setwd(&quot;../analysis&quot;)
# 
# |--------------------------------------|
# | Welcome to FINEMAP v1.4.1            |
# |                                      |
# | (c) 2015-2022 University of Helsinki |
# |                                      |
# | Help :                               |
# | - ./finemap --help                   |
# | - www.finemap.me                     |
# | - www.christianbenner.com            |
# |                                      |
# | Contact :                            |
# | - finemap@christianbenner.com        |
# | - matti.pirinen@helsinki.fi          |
# |--------------------------------------|
# 
# --------
# SETTINGS
# --------
# - dataset            : all
# - corr-config        : 0.95
# - n-causal-snps      : 5
# - n-configs-top      : 50000
# - n-conv-sss         : 100
# - n-iter             : 100000
# - n-threads          : 1
# - prior-k0           : 0
# - prior-std          : 0.05 
# - prob-conv-sss-tol  : 0.001
# - prob-cred-set      : 0.95
# 
# ------------
# FINE-MAPPING (1/1)
# ------------
# - GWAS summary stats               : sim2.z
# - SNP correlations                 : ../data/small_data_11.ld_refout_file.refout.ld
# - Causal SNP stats                 : sim2.snp
# - Causal configurations            : sim2.config
# - Credible sets                    : sim2.cred
# - Log file                         : sim2.log_sss
# 
# 
- Updated prior SD of effect sizes : 0.05 0.0906 0.164 0.297 
# 
# - Number of GWAS samples           : 800
# - Number of SNPs                   : 1001
# - Prior-Pr(# of causal SNPs is k)  : 
#   (0 -&gt; 0)
#    1 -&gt; 0.583
#    2 -&gt; 0.291
#    3 -&gt; 0.097
#    4 -&gt; 0.0242
#    5 -&gt; 0.00483
# - Regional SNP heritability        : 0.986 (SD: 0.000581 ; 95% CI: [0.984,0.987])
# - Log10-BF of &gt;= one causal SNP    : 185
# - Post-expected # of causal SNPs   : 5
# - Post-Pr(# of causal SNPs is k)   : 
#   (0 -&gt; 0)
#    1 -&gt; 0
#    2 -&gt; 0
#    3 -&gt; 4.93e-126
#    4 -&gt; 4.06e-113
#    5 -&gt; 1
# - Run time                         : 0 hours, 0 minutes, 23 seconds</code></pre>
<p>With the out-of-sample LD estimate, FINEMAP gives a very different result: it predicts 5 causal SNPs (which is the maximum we allowed), all the CSs contain a single SNP (except the last CS, which has 2 SNPs), and none of the CSs contain a true causal SNP.</p>
<pre class="r"><code>cat(readLines(&quot;../output/sim2.cred5&quot;),sep = &quot;\n&quot;)
# # Post-Pr(# of causal SNPs is 5) = 1
# #log10bf 600.715 NA 531.267 NA 135.593 NA 133.711 NA 117.309 NA
# #min(|ld|) 1 NA 1 NA 1 NA 1 NA 1 NA
# #mean(|ld|) 1 NA 1 NA 1 NA 1 NA 1 NA
# #median(|ld|) 1 NA 1 NA 1 NA 1 NA 1 NA
# index cred1 prob1 cred2 prob2 cred3 prob3 cred4 prob4 cred5 prob5
# 1 765 1 949 1 592 1 583 1 389 0.5
# 2 NA NA NA NA NA NA NA NA 405 0.5</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.5.1 (2018-07-02)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: Scientific Linux 7.4 (Nitrogen)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
# [1] susieR_0.12.07    data.table_1.12.0
# 
# loaded via a namespace (and not attached):
#  [1] tidyselect_1.1.0   xfun_0.31          bslib_0.3.1        purrr_0.3.4       
#  [5] lattice_0.20-38    colorspace_1.3-2   vctrs_0.3.6        generics_0.0.2    
#  [9] htmltools_0.5.2    yaml_2.2.0         utf8_1.1.4         rlang_0.4.10      
# [13] mixsqp_0.3-46      jquerylib_0.1.4    later_0.7.5        pillar_1.5.0      
# [17] glue_1.4.2         DBI_1.0.0          matrixStats_0.54.0 lifecycle_1.0.0   
# [21] plyr_1.8.4         stringr_1.3.1      munsell_0.5.0      gtable_0.2.0      
# [25] workflowr_1.7.0    evaluate_0.15      knitr_1.39         fastmap_1.1.0     
# [29] httpuv_1.4.5       irlba_2.3.3        fansi_0.4.0        highr_0.7         
# [33] Rcpp_1.0.7         promises_1.0.1     backports_1.1.2    scales_1.0.0      
# [37] jsonlite_1.6       fs_1.5.0           ggplot2_3.3.3      digest_0.6.18     
# [41] stringi_1.2.4      dplyr_1.0.5        rprojroot_1.3-2    grid_3.5.1        
# [45] tools_3.5.1        magrittr_1.5       sass_0.4.1         tibble_3.1.0      
# [49] crayon_1.3.4       whisker_0.3-2      pkgconfig_2.0.2    ellipsis_0.3.2    
# [53] Matrix_1.2-15      rmarkdown_2.14     reshape_0.8.8      R6_2.3.0          
# [57] git2r_0.26.1       compiler_3.5.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
